packages/deployment/lib/lambda.js

```diff
@@ -27,6 +29,28 @@ class UpdatedLambda extends Lambda {
     super(config);
     this.config = config;
   }
+
+  /**
+   * Makes use of yauzl.open's verification
+   * to validate the file referenced in lambda.local
+   * is a valid zip archive
+   *
+   * @param {Object} lambda - the lambda object
+   *
+   * @returns {Promise} promise resolution indicates success, will reject if
+   *                    zipfile cannot be opened
+   **/
+  validateZipFile(lambda) {
+    return new Promise((resolve, reject) => {
+      yauzl.open(lambda.local, (err, _zipfile) => {
+        if (err) {
```

Comment:
This should be:

```
if (err) return reject(err);
return resolve();
```

If you don't return the rejection, it will continue on in the function and call `resolve()` as well.  Speaking from experience, that's an annoying bug to track down.

Reply:
On second thought, if you add `const util = require('util');` to the top of this file, this entire function can be rewritten as:

```
return (util.promisify(yauzl.open))(lambda.local);
```

Reply:
@yjpa7145  Nice, I like that *far* better.  TIL.   Thanks!